plugins/org.polarsys.capella.scenario.editor/src/org/polarsys/capella/scenario/editor/EmbeddedEditorInstance.java

```diff
@@ -21,6 +22,7 @@
 public class EmbeddedEditorInstance {
   private static EmbeddedEditor eEditor;
   private static Scenario associatedScenarioDiagram;
```

Comment:
Since DDiagram holds the info of Scenario you can remove var associatedScenarioDiagram.